code/modules/mob/mob_helpers.dm

```diff
@@ -131,6 +131,71 @@ GLOBAL_LIST_INIT(limb_types_by_name, list(
 		index++
 	return output_message
 
+/**
+ * Summary: proc that parses an html input string and scrambles the non-html string contents.
+ *
+ * Arguments:
+ * * message - an html string value to be parsed and modified.
+ *
+ * Return:
+ * returns the parsed and modified html output with the text content being partially scrambled with asteriks
+ */
+/proc/stars_decode_html(message)
+	if(!length(message))
+		return
+
+	// boolean value to know if the current indexed element needs to be scrambled.
+	var/parsing_message = FALSE
+
+	// boolean values to know if we are currently inside a double or single quotation.
+	var/in_single_quote = FALSE
+	var/in_double_quote = FALSE
+
+	// string of what tag we're currently in
+	var/current_tag = ""
+	var/escaped_tag = FALSE
+
+	// string that will be scrambled
+	var/current_string_to_scramble = ""
+
+	// output string after parse
+	var/output_message = ""
+	for(var/character_index in 1 to length(message))
+		var/current_char = message[character_index]
+
+		// Apparent edge case safety, we only want to check the < and > on the edges of the tag.
+		if(!parsing_message)
+			if(current_char == "'")
+				in_single_quote = !in_single_quote
+			if(current_char == "\"")
+				in_double_quote = !in_double_quote
+			if(in_single_quote || in_double_quote)
+				output_message += current_char
+				continue
+
+		if(current_char == ">")
+			parsing_message = TRUE
+			output_message += current_char
+			if(current_tag == "<style" || findtext(current_tag, "<style ") == 1) // findtext because HTML doesn't care about anything after whitespace
+				escaped_tag = TRUE
+			else if(escaped_tag && (current_tag == "</style" || findtext(current_tag, "</style ") == 1)) // 1 for findtext because we only care about the start of the string matching
+				escaped_tag = FALSE
```

Comment:
```suggestion
			if(findtext(current_tag, "<style") == 1) // findtext because HTML doesn't care about anything after whitespace
				escaped_tag = TRUE
			else if(escaped_tag && findtext(current_tag, "</style") == 1) // 1 for findtext because we only care about the start of the string matching
				escaped_tag = FALSE
```
Case insensitive and just cleaner.